llvm/lib/Target/X86/GISel/X86LegalizerInfo.cpp

```diff
@@ -521,11 +521,10 @@ X86LegalizerInfo::X86LegalizerInfo(const X86Subtarget &STI,
 
   // todo: vectors and address spaces
   getActionDefinitionsBuilder(G_SELECT)
-       .legalFor({{s8, s32}, {s16, s32}, {s32, s32}, {s64, s32},
-                  {p0, s32}})
-    .widenScalarToNextPow2(0, /*Min=*/8)
-    .clampScalar(0, s8, sMaxScalar)
-    .clampScalar(1, s32, s32);
+      .legalFor({{s8, s32}, {s16, s32}, {s32, s32}, {s64, s32}, {p0, s32}})
+      .widenScalarToNextPow2(0, /*Min=*/8)
+      .clampScalar(0, s8, sMaxScalar)
+      .clampScalar(1, s32, s32);
```

Comment:
Could we promote i8 cases on targets with CMOV?

Reply:
In that case, you add:
```
bool HasCMOV = STI.canUseCMOV();
```
to the top of the file.